code/modules/mob/dead/new_player/preferences_setup.dm

```diff
@@ -48,7 +52,7 @@
 
 ///Setup a hardcore random character and calculate their hardcore random score
 /datum/preferences/proc/hardcore_random_setup(mob/living/carbon/human/character, antagonist)
-	random_character(TRUE, antagonist)
+	random_character(FALSE, antagonist)

```

Comment:
was this the fix? from a quick glance, this looks like it will allow people to de-fullrandom their character by picking their gender.

Reply:
That argument doesn't accept "true", it's supposed to take the gender you're using to override. I assume hardcore sets the random gender flag elsewhere, but just in case it doesn't, I'll do another gender randomization on top